### ISSUE TITLE
linting feature request: no single star comments

### ISSUE THREAD

**Issue by Carus11** [sasjs/lint#95]
Could I request a rule to ensure there are none of those single star comments, basically

```
* This is what my code does 
```
Fails, and:
```
/* This is what my code does */
``` 
Passes

**Comment by allanbowe** [comment]
Hey Carus!  I've been thinking abou this.  It doesn't sound easy.  These are the initial test cases I came up with - can you think of any more?

```sas
/**
 * Examples of *message; format that SHOULD be caught
 */

/* 1. regular comment */
* some text;

/* 2. multiline comment */
*
multiline
text;

/* multiple asterisks - should start on first */
   *---------------------------------------*
   |  This uses one comment statement      |
   |           to draw a box.              |
   *---------------------------------------*;
   

/* 3. comment at end of statement */
libname x (work); * some text;


/* 4. comment in middle of statement */
run; *walk
; run;

/* 5. comment at start of statement */
*; run;



/**
 * Examples of *message; format that should NOT be caught
 * (this was the first)
 */

/* 2. In a string literal */
data _null_;
  string='no*no;no';
  string="not*this;either";
run;

/* 3. In a %str() block */
%let x=%str(afraid *not; sir);

/* 4. In an nrstr() block */
%let y=%nrstr(try*again;please);

/* 5. In a %put statement */
%put not * today;;

/* 6. As a macro comment */
%* this one is totally normal; 

/* 7. in a LUA proc */
proc lua;
submit;
 -- hi *mum;
endsubmit;
run;
```

**Comment by Carus11** [comment]
Oh wow thanks Allan,
I'll have some of our devs have a look at your cases. I just realized now my example was wrong because I didnt terminate the comment with a semicolon. I am going to leave it that way as that is the very mistake I find sneaks into our code base time and time again when we comment using the single asterisk method.

**Comment by allanbowe** [comment]
I'm wondering if we will need to build a tokenizer to make this work!

**Comment by allanbowe** [comment]
@all-contributors add @Carus11 for ideas

**Comment by krishna-acondy** [comment]
Hello @Carus11 @allanbowe Thanks for the idea and your input. 🙌🏽 
I'll take an initial crack at implementing this at some point next week.
